gcn_kafka/core.py

```diff
@@ -53,6 +55,32 @@ def get_config(mode, config, **kwargs):
     set_oauth_cb(config)
     return config
 
+env_key_splitter = re.compile(r'_+')
+replacement_dict = {'_': '.', '__': '-', '___': '_'}
+
+def replacement(match: re.Match) -> str:
+    text = match[0]
+    return replacement_dict.get(text) or text
+
+def kafka_config_from_env(env: dict[str, str], prefix: str) -> dict[str, str]:
```

Comment:
* Make `KAFKA_` the default prefix.
* Rename to `config_from_env`, because now that this lives in the `gcn_kafka` package, it should be obvious that this function returns Kafka configuration.
* Make the `env` argument optional, so that if it is omitted it is read from `os.environ`.

```suggestion

def config_from_env(env: dict[str, str], prefix: str = 'KAFKA_') -> dict[str, str]:
```